### ISSUE TITLE
Respawn doesn't work with Papa 

### ISSUE THREAD

**Issue by cgarciaarano** [circus-tent/circus#946]
When `use_papa = True`, processes are not respawning.

watcher.ini:

```ini
[watcher:test_papa_respawn]
uid = root
gid = root
cmd = sleep 5
numprocesses = 1
use_papa = True
autostart = True
respawn = True
```

After starting the watcher:
```
(circusctl) stats test_papa_respawn
test_papa_respawn:
8261: 8261  sleep root 0 348K 4M 0.0 0.0 0:00.0
```

After 5 seconds, the process die and it's not respawned:
```
(circusctl) stats test_papa_respawn
test_papa_respawn:
8261: No such process (stopped?)
```

It also happens if the process is killed by an external signal.

This issue can be a feature request, but I think that it should be warned in documentation. It says that Papa does not restart processes, but someone could understand (as I did) that circus will handle the restarting.

**Comment by k4nar** [comment]
Indeed, I think this is by design and not a bug per se. I'm sure @scottkmaxwell has a great explanation :) .

**Comment by scottkmaxwell** [comment]
Papa does not restart processes, but the intention is that circus will see that the process has quit and that it will do the respawn. So this is a bug. It sounds like I have 2 or 3 things to dig into on the papa front. I've been completely swamped but I'll try to carve out some time over the break to get these fixed.

**Comment by scottkmaxwell** [comment]
I am having a look at this right now. It looks like papa is reporting the correct state but circus is not understanding that the process has been stopped. I wrote the code on both sides so I just have to remember how it works.

**Comment by scottkmaxwell** [comment]
Got it. This was a special case I never tested. The papa watcher reads the proxies output to get stdout, stderr, and the exit status. These are preserved until they are read. In this case, stdout and stderr were not redirected, so the watcher was never pumped and the old process never cleared. I added a special-case check for no redirect and just remove stopped processes in those cases.

**Comment by scottkmaxwell** [comment]
Fixed in #953
